### ISSUE TITLE
[Bug]: XMLHttpRequest onreadystatechange not working for requests

### ISSUE THREAD

**Issue by JpEngelbrecht89** [ionic-team/capacitor#7456]
### Capacitor Version

💊   Capacitor Doctor  💊 

Latest Dependencies:

  @capacitor/cli: 6.0.0
  @capacitor/core: 6.0.0
  @capacitor/android: 6.0.0
  @capacitor/ios: 6.0.0

Installed Dependencies:

  @capacitor/ios: not installed
  @capacitor/cli: 6.0.0
  @capacitor/android: 6.0.0
  @capacitor/core: 6.0.0

[success] Android looking great! 👌

### Other API Details

```Shell
npm: 10.2.4
node: v20.11.0
```

### Platforms Affected

- [ ] iOS
- [X] Android
- [X] Web

### Current Behavior

After upgrading to @capacitor/android version 6.0.0, requests seem to be intercepted by CapacitorHttp but request.onreadystatechange is not updating readyState (readystate is alway 0). This is happening in both iOS and Android.

### Expected Behavior

CapacitorHttp intercepts all requests like it did with Capacitor v5

### Project Reproduction

https://github.com/JpEngelbrecht89/capacitor-http-error

### Additional Information

Think its similar to this issue [7417](https://github.com/ionic-team/capacitor/issues/7417)

This stared happening from version 5.7.1

**Comment by jcesarmobile** [comment]
yes, it's a duplicate of https://github.com/ionic-team/capacitor/issues/7417
